simpleclient/src/main/java/io/prometheus/client/CKMSQuantiles.java

```diff
@@ -185,30 +267,37 @@ private boolean insertBatch() {
             count++;
         }
 
+        // To insert a new item, v, we find i such that vi < v ≤ vi+1, 
         ListIterator<Item> it = sample.listIterator();
         Item item = it.next();
 
-        for (int i = start; i < bufferCount; i++) {
+        // Keep track of the current rank by adding the g of each item. See also discussion in https://issues.apache.org/jira/browse/HBASE-14324
+        // Paper Section 3.1 on true rank: let r_i = Sum_{j=1}^{i−1} g_j
+        int currentRank = item.g;
+
+        for (int i = start; i < bufferIdx; i++) {
+            // item to be inserted
             double v = buffer[i];
+            // find the item in the samples that is bigger than our v.
             while (it.nextIndex() < sample.size() && item.value < v) {
```

Comment:
Not your code, but I guess `it.nextIndex() < sample.size()` could be replaced with `it.hasNext()`.